### PR TITLE
Documentation of base widgets

### PR DESCRIPTION
**Pull request by tonybaloney**
* Add both undocumented and inherited members to the autodoc references for widgets. This will expose the base widget methods and the un-docstringed methods
* Add some explanation to the basic widgets and a more detailed description to the box reference
* Add an intro to the app class

Still needs #101 to be fixed before most of this will show on RTD.

### CODE REVIEW

**Comment by tonybaloney** [comment]
How does an optioncontainer relate to a select widget?

It looks kind of obvious, a list of tuples (label, selection) are sent to an option container?

**Comment by tonybaloney** [comment]
Another round of docstrings, docs and updates. I have a few questions about some of the specific containers. Like what is the difference between optioncontainer and selection? @freakboy3742

**Comment by tonybaloney** [comment]
this is done now. So much for 1 change per PR!

File: docs/reference/widgets/app.rst [hunk 1]
```diff
@@ -3,5 +3,36 @@ Application
 
 The app is the main entry point and container for the Toga GUI.
 
+Usage
+-----
+
+The app class is used by instantiating with a name, namespace and callback to a startup delegate which takes 1 argument of the app instance.
```
Comment (freakboy3742) [inline docs/reference/widgets/app.rst]: This is correct; but you can also subclass App and define a `startup()` method.

File: docs/supported_platforms.rst [hunk 2]
```diff
@@ -1,59 +1,64 @@
...
-    |Image             |no     |no    |no   |no |yes  |no |no |
-    +------------------+-------+------+-----+---+-----+---+---+
-    |Label             |no     |no    |yes  |yes|yes  |yes|no |
-    +------------------+-------+------+-----+---+-----+---+---+
-    |EXPANDING_SPACER  |no     |no    |no   |yes|yes  |no |no |
-    +------------------+-------+------+-----+---+-----+---+---+
+    +------------------+-----+-----+-----+-------+------+----+-----+
+    |    Component     | gtk |win32|cocoa|android|django|web | iOS |
+    +==================+=====+=====+=====+=======+======+====+=====+
+    |Icon              ||yes|||no| ||yes|||no|   ||no|  ||no|||no| |
```
Comment (freakboy3742) [inline docs/supported_platforms.rst]: Would it be possible to make these link off to the relevant reference pages?

Comment (tonybaloney) [inline docs/supported_platforms.rst]: done in https://github.com/pybee/toga/pull/102/commits/b0a5fbf68e6b1d875c8767f325ffcd54e2e22688

File: docs/reference/widgets/app.rst (hunk 1 above)
Comment (tonybaloney) [inline docs/reference/widgets/app.rst]: done in https://github.com/pybee/toga/pull/102/commits/45a92233b0d194afaee80cde9129ae7d8230ecf3

**Comment by tonybaloney** [comment]
updated to reflect suggested changes @freakboy3742

**Pull request merged**